csharp/private/wrappers/dotnet.cc

```diff
@@ -50,14 +63,34 @@ int main(int argc, char** argv) {
       evprintf("DOTNET_CLI_TELEMETRY_OPTOUT", "1"),  // disable telemetry
   };
 
+  auto extra_count =
+    !!(std::strlen(kArgv1) != 0) +
+    !!(std::strlen(kArgv2) != 0);
+
+  auto dotnet_argv = new char*[extra_count + argc + 1];
+
   // dotnet wants this to either be dotnet or dotnet.exe but doesn't have a
   // preference otherwise.
-  auto dotnet_argv = new char*[argc + 1];
-  dotnet_argv[0] = (char*)"dotnet";
+  dotnet_argv[0] = const_cast<char*>("dotnet");
+
+  // Make sure to hold a reference to these std::string.
```

Comment:
Because of the pointers from c_str?

Reply:
`Rlocation` returns an `std::string` and yeah, the output of `c_str()` is coupled to that lifetime.